pkg/engine/jmespath/functions.go

```diff
@@ -925,6 +938,71 @@ func jpRound(arguments []interface{}) (interface{}, error) {
 	return rounded, nil
 }
 
+func jpFormatNumber(arguments []interface{}) (interface{}, error) {
+	funcCall, err := validateArg(formatNumber, arguments, 0, reflect.String)
+	if err != nil {
+		return nil, err
+	}
+	arg := arguments[1:]
+	var formatted string
+
+	switch funcCall.String() {
+	case compare:
```

Comment:
Why is this switch case needed? I am a bit lost why we are doing this if we only expect a number?

Reply:
Umm should I use an if else instead? if (for round) and else for all the others? and have a validateArg to check the return type of the calling function?

Reply:
So depending on the string value, that particular function will be called and since only specific functions return int not all functions have been included in the switch case. I hope its clear, offcourse if you have a better solution let me know, Thanks !